technology/core-skills/tdd/inappropriate.md

```diff
@@ -1,11 +1,24 @@
 # Inappropriate TDD
 
-Graphical UIs are not suitable for TDD, they are usually defined in a markup language which can only be asserted for correctness by visually inspecting the rendered output.
+## Validation of correctness cannot be obtained via an automated check.
 
-Configurations are not suitable for TDD, their correctness can only be ascertained by testing the configuration in conjunction with whatever they are configuring.
+Sometimes you can find a proxy for "correctness" which will make you feel like TDD is possible.
+If you aren't feeling confident that your tests are guarding against regression, this is a signal that you shouldn't be using TDD.
 
-State machines are configuration? Unit testing is still important, but the discipline of TDD is not.
+Graphical UIs are an example of software not suitable for TDD, they are usually defined in a markup language which can only be asserted for correctness by visually inspecting the rendered output.
+
+Configurations are also not suitable for TDD, their correctness can only be ascertained by testing the configuration in conjunction with whatever they are configuring.
+State machines are configuration. Unit testing is still important, but the discipline of TDD is not.
+
+## Testing feedback cycles are slow
+
+Some software is hard to test because of inadequate test tooling.
+A slow test driven development cycle is painful and unproductive.
+Appropriate TDD makes your development cycle faster via fewer defects, and testable code.
+Weight up the benefits of test driven development vs the cost of delaying your development in this scenario.
+
+Investing time into researching how to create fast feedback cycles can have huge gains in productivity.
 
 ## Marking Scheme
 
-Does not TDD markup.
+* Does not TDD markup.
```

Comment:
What does this mean?

Reply:
Maybe needs to be Mark{down,up} :D 

I had HTML in mind, you shouldn't TDD configuration and HTML is often like a configuration.